src/Raven.Client/Documents/Session/Operations/GetRevisionOperation.cs

```diff
@@ -98,6 +98,9 @@ public List<T> GetRevisionsFor<T>()
 
         public List<MetadataAsDictionary> GetRevisionsMetadataFor()
         {
+            if (_result?.Results == null || _result.Results.Length == 0)
```

Comment:
when can this happened?

Reply:
Right, the fix should be in `ShardedRevisionsHandlerProcessorForGetRevisions.GetRevisionsAsync()`. Changed :-)